packages/peregrine/lib/talons/ProductFullDetail/useProductFullDetail.js

```diff
@@ -178,6 +181,12 @@ export const useProductFullDetail = props => {
     );
 
     const [{ cartId }] = useCartContext();
+    const [{ isSignedIn, currentUser }] = useUserContext();
```

Comment:
Rather than using `currentUser` to access data, you should make queries and mutations.

In this case, you would likely use `useLazyQuery` to fetch user data if the user is signed in.

Reply:
that mean each time we come to another product then call that query 1 time? why we don't use id as my code? I think it's better

Reply:
The way ApolloClient works with the cache is configurable. We have opted, in Venia, to use `cache-and-network` for first render, and `cache-first` for subsequent renders (assuming no refresh).

What this means is that the first time a user visits a page, they see data from the cache, if present. Then ApolloClient makes a network call to the GQL server to fetch up-to-date data, and replaces the older data in the cache. The result is an immediately visible page (based on cached data) which is then visually updated with any new information that may come from the back end.

Applying that example to the wishlist UX: Let's say we have a component `AddToWishlistButton` on the product page. This button/input contains a dropdown of wishlists. The first time the page loads, we make a query to get a list of the customer's wishlists to render in the button. Then if we go to a different product page, the data for that button is already cached, so the button is immediately visible. In a similar vein, if a user visits their account wishlist page we would have their wishlists already in cache, so when they visit the product page we can immediately render them (and make a true-up network call behind the scenes).

Reply:
That mean should I create new component `AddToWishlistButton` & that talon then how can I get quantity of form?